### ISSUE TITLE
I always get error when i run a ql file using query-overpass

### ISSUE THREAD

**Issue by sunilrx** [perliedman/query-overpass#22]
Hi
I always get error when i run a ql file. the error is 
"**/usr/bin/env: ‘node\r’: No such file or directory**"

I am new to node, please help.

**Comment by perliedman** [comment]
It sounds like you have Node installed in such a way that `env` doesn't find it.

If you look at the first line of `cli.js`, it's

```
#!/usr/bin/env node
```

which is telling your system (or its `env` command) to look up where the `node` executable is, and run the rest of the file with it. Apparently, your system does not find `node`, but I don't know why. Which operating system/version etc. is this on?

**Comment by sunilrx** [comment]
I use
Distributor ID: Ubuntu
Description:    Ubuntu 16.10
Release:        16.10
Codename:       yakkety

**Comment by tyrasd** [comment]
@sunilrx try installing [nodejs-legacy](http://packages.ubuntu.com/en/yakkety/nodejs-legacy) (`sudo apt-get install nodejs-legacy`), which adds the necessary symlink to the `node` command which is missing in Debian based Linux systems by default (for legacy reasons).

**Comment by perliedman** [comment]
@sunilrx presumably you installed Node with Ubuntu's package manager? In that case, Ubuntu installs node as `nodejs`, not `node`, in contrast to the rest of the world. This is highly annoying.

You can fix this by following the instructions here, there's also some context about this problem: https://askubuntu.com/a/521571/2393

(Or just follow @tyrasd's advice, since he beat me :smile: )

**Comment by sunilrx** [comment]
@tyrasd thanks, but i have tried installing the nodejs-legacy. but i get the following error

**The following packages have unmet dependencies:
 nodejs-legacy : Depends: nodejs (>= 0.6.19~dfsg1-3~) but it is not going to be installed
E: Unable to correct problems, you have held broken packages**.

don't know how to move forward. Also i have tried 
**ln -s /usr/bin/nodejs /usr/bin/node**

Still not working. But i have successfully installed **geojsonio-cli** ;-)

Any idea ? Thanks for you help

**Comment by tyrasd** [comment]
> successfully installed geojsonio-cli

Ok. Wait.

> "/usr/bin/env: ‘node\r’

Note the `\r` here. That shouldn't be there. And as far as I can see it's also not in the code of the git repository here on github:

```
$ hd cli.js | head -n 2
00000000  23 21 2f 75 73 72 2f 62  69 6e 2f 65 6e 76 20 6e  |#!/usr/bin/env n|
00000010  6f 64 65 0a 0a 76 61 72  20 63 6f 6e 63 61 74 20  |ode..var concat |
```

But it's apparently there in the code that's published via npm:

```
$ hd cli.js | head -n 2
00000000  23 21 2f 75 73 72 2f 62  69 6e 2f 65 6e 76 20 6e  |#!/usr/bin/env n|
00000010  6f 64 65 0d 0a 0d 0a 76  61 72 20 63 6f 6e 63 61  |ode....var conca|
```

@perliedman - could it be that there was any issue before/while running `npm publish` that could have led to this?

@sunilrx - for the time being, try to install the previous version of query-overpass from npm: e.g. `sudo npm install -g query-overpass@1.2.0`

**Comment by sunilrx** [comment]
@tyrasd thanks. that works.
1. but i have this error when i run **query-overpass query.ql**

**{ message: 'Request failed: HTTP 400', statusCode: 400 }**

2, Also one question, if i have the overpass API and db cloned in my server, can i request and get map output data(json) from my server using the **query-overpass** ?
It will be very helpful, as i have just started working on OSM

**Comment by perliedman** [comment]
@sunilrx:

1. AFAIK, a 400 response means the query is invalid (HTTP 400 is "Bad request") - you can perhaps verify your query in Overpass Turbo: https://overpass-turbo.eu/
2. See [the CLI docs](https://github.com/perliedman/query-overpass#cli), you want the `--overpass-url` option

**Comment by sunilrx** [comment]
Thanks a lot @perliedman and @tyrasd for help

**Comment by perliedman** [comment]
@tyrasd regarding the carriage return, I think I might have published the last version from Windows (I'm dual booting at work, so sometimes it's convenient to do quick things without rebooting), and I guess that's why it's there.

However, 1.3.0 still gets an "works on my machine" badge, I can successfully run it on Ubuntu 16.04. Weird.

**Comment by tyrasd** [comment]
> works on my machine

Strange, this is what I'm getting (I'm currently working on a Ubuntu 16.10 machine):

```
$ node -v
v7.5.0
$ sudo npm install -g query-overpass
npm WARN deprecated geojson-area@0.1.0: This module is now under the @mapbox namespace: install @mapbox/geojson-area instead
/usr/local/bin/query-overpass -> /usr/local/lib/node_modules/query-overpass/cli.js
/usr/local/lib
└─┬ query-overpass@1.3.0 
  └─┬ xhr@2.4.0 
    ├─┬ global@4.3.2 
    │ ├─┬ min-document@2.19.0 
    │ │ └── dom-walk@0.1.1 
    │ └── process@0.5.2 
    ├── is-function@1.0.1 
    ├─┬ parse-headers@2.0.1 
    │ ├── for-each@0.3.2 
    │ └── trim@0.0.1 
    └── xtend@4.0.1 

$ query-overpass 
/usr/bin/env: ‘node\r’: No such file or directory
```

**Comment by sunilrx** [comment]
Same as mine at the begnning :-)

**Comment by perliedman** [comment]
I published 1.4.0 from Ubuntu just now, so this should hopefully be resolved if you install latest!

**Comment by tyrasd** [comment]
It now works for me again. Thanks!
